### ISSUE TITLE
Three-variable grouping in Python

### ISSUE THREAD

**Issue by jhofman** [microsoft/datamations#132]
@chisingh will work on a three-variable grouping example.

@sharlagelfand, can you add the spec from the shiny app example here?

**Comment by sharlagelfand** [comment]
@chisingh [Here](https://github.com/microsoft/datamations/blob/main/sandbox/penguins_three_groups.json) is an example spec grouping the penguin data by three grouping variables and getting the mean bill_length_mm.

It uses the [palmerpenguins](https://github.com/allisonhorst/palmerpenguins) dataset, looks like there's also a [python version of the package](https://github.com/mcnakhaee/palmerpenguins)! The datamations package itself doesn't contain the penguins data set, but rather shows loading `palmerpenguins` and using the data from there, so might be good to try out using an external data set in the python version too.

R code for the operation:

```r
library(palmerpenguins)
library(dplyr)

penguins %>%
  group_by(species, island, sex) %>%
  summarize(mean = mean(bill_length_mm))
```

**Comment by sharlagelfand** [comment]
Just a thought for testing, it would be good to generate the groups in a different order (e.g. species, island, sex vs island, sex, species) and also test generating the mean of different columns (there's bill length, bill depth, etc).

**Comment by chisingh** [comment]
Closed by #162
